Release/src/http/client/http_client_curl.cpp

```diff
@@ -0,0 +1,2379 @@
+    {WINHTTP_CALLBACK_STATUS_FLAG_CERT_DATE_INVALID,
+     "WINHTTP_CALLBACK_STATUS_FLAG_CERT_DATE_INVALID SLL certificate is expired."},
+    {WINHTTP_CALLBACK_STATUS_FLAG_SECURITY_CHANNEL_ERROR,
+     "WINHTTP_CALLBACK_STATUS_FLAG_SECURITY_CHANNEL_ERROR internal error."},
+};
+
+std::string generate_security_failure_message(std::uint32_t flags)
+{
+    std::string result("SSL Error:");
+    for (const auto& message : g_security_failure_messages)
+    {
+        if (flags & message.flag)
+        {
+            result.push_back(' ');
+            result.append(message.text);
+        }
+    }
+
+    return result;
+}
+
+} // unnamed namespace
+namespace web
+{
+namespace http
+{
+namespace client
+{
+namespace details
+{
+// Helper function to query for the size of header values.
+static void query_header_length(HINTERNET request_handle, DWORD header, DWORD& length)
+{
+    WinHttpQueryHeaders(request_handle,
+                        header,
+                        WINHTTP_HEADER_NAME_BY_INDEX,
+                        WINHTTP_NO_OUTPUT_BUFFER,
+                        &length,
+                        WINHTTP_NO_HEADER_INDEX);
+}
+
+// Helper function to get the status code from a WinHTTP response.
+static http::status_code parse_status_code(HINTERNET request_handle)
+{
+    DWORD length = 0;
+    query_header_length(request_handle, WINHTTP_QUERY_STATUS_CODE, length);
+    utility::string_t buffer;
+    buffer.resize(length);
+    WinHttpQueryHeaders(request_handle,
+                        WINHTTP_QUERY_STATUS_CODE,
+                        WINHTTP_HEADER_NAME_BY_INDEX,
+                        &buffer[0],
+                        &length,
+                        WINHTTP_NO_HEADER_INDEX);
+    return (unsigned short)atoi((LPCTSTR)(buffer.c_str()));
+}
+
+// Helper function to trim leading and trailing null characters from a string.
+static void trim_nulls(utility::string_t& str)
+{
+    size_t index;
+    for (index = 0; index < str.size() && str[index] == 0; ++index)
+        ;
+    str.erase(0, index);
+    for (index = str.size(); index > 0 && str[index - 1] == 0; --index)
+        ;
+    str.erase(index);
+}
+
+// Helper function to get the reason phrase from a WinHTTP response.
+static utility::string_t parse_reason_phrase(HINTERNET request_handle)
+{
+    utility::string_t phrase;
+    DWORD length = 0;
+
+    query_header_length(request_handle, WINHTTP_QUERY_STATUS_TEXT, length);
+    phrase.resize(length);
+    WinHttpQueryHeaders(request_handle,
+                        WINHTTP_QUERY_STATUS_TEXT,
+                        WINHTTP_HEADER_NAME_BY_INDEX,
+                        &phrase[0],
+                        &length,
+                        WINHTTP_NO_HEADER_INDEX);
+    // WinHTTP reports back the wrong length, trim any null characters.
+    trim_nulls(phrase);
+    return phrase;
+}
+
+
+static void parse_headers_string(_Inout_z_ TCHAR* headersStr, web::http::http_headers& headers)
```

Comment:
All the places that say TCHAR need to say utility::char_t (or similar).

Reply:
done